src/js/client/editprefset-prefSettingAdjuster-widget.js

```diff
@@ -0,0 +1,222 @@
+            path: "current.value",
+            rules: {
+                domToModel: {
+                    "": {
+                        transform: {
+                            type: "gpii.devpmt.prefSettingAdjuster.binderToValueInput",
+                            inputPath: ""
+                        }
+                    }
+                },
+                modelToDom: {
+                    "": {
+                        transform: {
+                            type: "gpii.devpmt.prefSettingAdjuster.valueInputToBinder",
+                            inputPath: ""
+                        }
+                    }
+                }
+            }
+        }
+    },
+    templates: {
+        initial: "editprefset-prefSettingAdjuster-widget"
+    },
+    invokers: {
+        renderInitialMarkup: {
+            func: "{that}.renderMarkup",
+            args: ["initial", "{that}.options.templates.initial", "{that}.model"]
+        },
+        saveUpdateValue: {
+            funcName: "gpii.devpmt.saveUpdateValue",
+            args: ["{that}", "{editPrefs}"]
+        },
+        watchInputKeys: {
+            funcName: "gpii.devpmt.watchInputKeys",
+            args: ["{that}", "{arguments}.0"]
+        },
+        updateBlankDisabling: {
+            funcName: "gpii.devpmt.updateBlankDisabling",
+            args: ["{that}"]
+        }
+    },
+    listeners: {
+        onMarkupRendered: [{
+            "this": "{that}.dom.okButton",
+            "method": "click",
+            args: ["{that}.saveUpdateValue"]
+        },
+        {
+            "this": "{that}.dom.valueInput",
+            "method": "keypress",
+            args: ["{that}.watchInputKeys"]
+        },
+        {
+            funcName: "fluid.focus",
+            args: ["{that}.dom.valueInput"]
+        },
+        {
+            func: "{that}.updateBlankDisabling"
+        }]
+    },
+    modelListeners: {
+        "current.blank": {
+            func: "{that}.updateBlankDisabling"
+        }
+    }
+});
+
+/**
+ * Updates the UI based on whether Blank is selected.
+ * If blank is selected, then the rest of the editing
+ * apparatus is disabled.
+ */
+gpii.devpmt.updateBlankDisabling = function (that) {
+    var disable = that.model.current.blank;
+    that.dom.locate("valueInput").prop("disabled", disable);
+};
+
+/**
+ * watchInputKeys - Watches value input and saves
+ * on Enter.
+ *
+ * @param (Object) that - prefSettingAdjuster
+ * @param (Object) e - jQuery Event
+ */
+gpii.devpmt.watchInputKeys = function (that, e) {
+    // Save the value when the enter key is pressed
+    if (e.keyCode === 13) {
+        that.saveUpdateValue();
+    };
+};
+
+/**
+ * Save/Update using the value the user has input. This is
+ * typically caused by clicking the Ok button.
+ */
+gpii.devpmt.saveUpdateValue = function (that, devpmt) {
+    that.applier.change("active", false);
+    var segs = ["contexts", that.model.current.context, "preferences"];
```

Comment:
The term "context" has been globally cursed for this purpose. The things inside a preferences document ("safe") have now come to be named "preference sets", presumably indexed by something called a "preference set name"

Reply:
bump.

Reply:
I understand that in comments, documentation, and elsewhere what we have now are referred to as preference sets, but this code above, and in most of the references to the internal PPT data model, we are editing paths for documents whose structures are defined in GPII/universal.  As of now, in universal, the data path for a preference set is:

```
preferences -> flat -> contexts -> PrefsSetObjects
```

See https://github.com/GPII/universal/blob/master/testData/preferences/alice.json5

until we go through a massive search replace in universal and it's dependent repos, I feel like I should use the same paths, since this tool is editing those same exact files.  When we change the file structure in universal, or at least define the updated schema version for that includes a different keys for the structure name, then those new key names could be tracked in the PPT as well.

Reply:
ELIMINATED